### PR TITLE
Set permissions for GitHub actions

### PR DESCRIPTION
**Pull request by nathannaveen**
- Included permissions for the action. https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

[Keeping your GitHub Actions and workflows secure Part 1: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)

 Restrict the GitHub token permissions only to the required ones; this way, even if the attackers will succeed in compromising your workflow, they won’t be able to do much.

Signed-off-by: nathannaveen <42319948+nathannaveen@users.noreply.github.com>

### CODE REVIEW

**Comment by AnomalRoil** [comment]
LGTM, could you add a `RELEASE_NOTES=n/a ` to your commit message for our changelog system?

**Review by AnomalRoil — APPROVED** [review]

File: .github/workflows/autorelease.yml [hunk 1]
```diff
@@ -9,6 +9,9 @@ on:
     tags:
       - 'v*'
           
+permissions:
+  contents: read
```
Comment (dominikschulz) [inline .github/workflows/autorelease.yml:13]: Is that really sufficient? I think our release process also needs to write. But I'm not 100% sure.

**Comment by dominikschulz** [comment]
Let's try it. If it fails we can undo the change easily.

**Review by dominikschulz — APPROVED** [review]
LGTM

**Pull request merged**